bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/handler/AbstractHomeConnectThingHandler.java

```diff
@@ -0,0 +1,1300 @@
+        if (isThingReadyToHandleCommand()) {
+            logger.debug("Handle \"{}\" command ({}). haId={}", command, channelUID.getId(), getThingHaId());
+            try {
+                Optional<HomeConnectApiClient> homeConnectApiClient = getApiClient();
+
+                if (command instanceof RefreshType) {
+                    updateChannel(channelUID);
+                } else if (command instanceof StringType && CHANNEL_BASIC_ACTIONS_STATE.equals(channelUID.getId())
+                        && homeConnectApiClient.isPresent() && getBridgeHandler().isPresent()) {
+                    HomeConnectApiClient apiClient = homeConnectApiClient.get();
+                    updateState(channelUID, new StringType(""));
+
+                    if (COMMAND_START.equalsIgnoreCase(command.toFullString())) {
+                        HomeConnectBridgeHandler homeConnectBridgeHandler = getBridgeHandler().get();
+                        // workaround for api bug
+                        // if simulator, program options have to be passed along with the desired program
+                        // if non simulator, some options throw a "SDK.Error.UnsupportedOption" error
+                        if (homeConnectBridgeHandler.getConfiguration().isSimulator()) {
+                            apiClient.startSelectedProgram(getThingHaId());
+                        } else {
+                            @Nullable
+                            Program selectedProgram = apiClient.getSelectedProgram(getThingHaId());
+                            if (selectedProgram != null) {
+                                apiClient.startProgram(getThingHaId(), selectedProgram.getKey());
+                            }
+                        }
+                    } else if (COMMAND_STOP.equalsIgnoreCase(command.toFullString())) {
+                        apiClient.stopProgram(getThingHaId());
+                    } else if (COMMAND_SELECTED.equalsIgnoreCase(command.toFullString())) {
+                        apiClient.getSelectedProgram(getThingHaId());
+                    } else {
+                        logger.debug("Start custom program. command={} haId={}", command.toFullString(),
+                                getThingHaId());
+                        apiClient.startCustomProgram(getThingHaId(), command.toFullString());
+                    }
+                } else if (command instanceof StringType && CHANNEL_SELECTED_PROGRAM_STATE.equals(channelUID.getId())
+                        && homeConnectApiClient.isPresent()) {
+                    homeConnectApiClient.get().setSelectedProgram(getThingHaId(), command.toFullString());
+                }
+            } catch (ApplianceOfflineException e) {
+                logger.debug("Could not handle command {}. Appliance offline. thing={}, haId={}, error={}",
+                        command.toFullString(), getThingLabel(), getThingHaId(), e.getMessage());
+                updateStatus(OFFLINE);
```

Comment:
In my opinion, the message 'Could not handle command...' in the Thing status is not appropriate. It is an error message as a result of an action. I think it is ok to set the status to `OFFLINE` without a message.

Reply:
If it can only fail if the device is offline, you could add "Communication timed out" or so, to the status detail message.

Will the Thing come online again, when it is reachable again?

Reply:
Yes will come online again. Will change it as suggested.